### PR TITLE
Aligns PR Template with Publish

### PR DESCRIPTION
**Pull request by dcyoung-dev**
## Context

Having different PR templates is confusing and we sometime miss checks that are detailed in Publish

## Changes proposed in this pull request

Aligned PR Template with the Publish repo https://github.com/DFE-Digital/publish-teacher-training/pull/4419

## Guidance to review

N/A

## Link to Trello card

N/A

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [ ] Inform data insights team due to database changes

### CODE REVIEW

**Review by CatalinVoineag — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (avinhurry) [inline .github/PULL_REQUEST_TEMPLATE.md:29]: Do people actually do this anymore? There was a period of time where we used to squash all our commits before merging to help maintain a 'cleaner' history but i'm not sure people have been following that.

Comment (CatalinVoineag) [reply .github/PULL_REQUEST_TEMPLATE.md:29]: I personally try to squash commits. I guess it's a bit of an art to know what to squash and what not to if you have a big PR and you try to have multiple commits to make it easier to review.

Comment (tomas-stefano) [reply .github/PULL_REQUEST_TEMPLATE.md:29]: My view is that we shouldn't approve if the commit history is not satisfied or not organised. A suggestion is to open a discussion about things to check on a PR before approving?

Comment (dcyoung-dev) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: I'd like to discuss this further - lets keep it in for now 👌

**Review by avinhurry — APPROVED** [review]

**Pull request merged**
